drivers/overlay/peerdb.go

```diff
@@ -244,13 +237,12 @@ func (d *driver) peerDbUpdateSandbox(nid string) {
 
 		peerOps = append(peerOps, op)
 	}
-	pMap.Unlock()
 
 	for _, op := range peerOps {
 		op()
 	}
+	pMap.Unlock()
```

Comment:
As discussed, holding a lock around the calls that adds the vtep neighbor entries needs to be done carefully.
am not against this change. But It needs some more eyes. @sanimej wdyt ?